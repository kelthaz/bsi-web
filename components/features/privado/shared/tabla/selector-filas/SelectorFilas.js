import PropTypes from 'prop-types';
import { useState } from 'react';

import Select from '../../../../../shared/select/Select';

import styles from './selector-filas.module.scss';

const SelectorFilas = ({ rowsPerPage, currentPage, totalRows, onChange }) => {

  const [value, setValue] = useState({label: rowsPerPage, value: rowsPerPage});

  const lastIndex = currentPage * rowsPerPage;
  const firstIndex = (lastIndex - rowsPerPage) + 1;

  const handleChange = (val) => {
    setValue(val);
    onChange(val.value);
  };

  return (
    <div className={styles.inline}>
      <span className={`color-gray-light ${styles['data-count']}`}>
        {firstIndex} - {lastIndex > totalRows ? totalRows : lastIndex} de {totalRows} datos
      </span>
      <span className={`color-gray py-1 ml-2 pl-2 ${styles['rows-per-page']}`}>Filas por p√°gina:</span>
      <div className="w-25 ">
        <Select
          name="filasPorPagina"
          showAlwaysErrors={false}
          size="smallb"
          label=""
          value={value}
          setValue={handleChange}
          touched
          setTouched={() => {}}
          items={[
            { value: 5, label: 5 },
            { value: 10, label: 10 },
            { value: 20, label: 20 },
            { value: 50, label: 50 }
          ]}
        />
      </div>
    </div>
  );
};

SelectorFilas.propTypes = {
  rowsPerPage: PropTypes.number.isRequired,
  currentPage: PropTypes.number.isRequired,
  totalRows: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired
};

export default SelectorFilas;
