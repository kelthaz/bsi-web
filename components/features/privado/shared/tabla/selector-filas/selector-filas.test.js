import { shallow } from 'enzyme';
import React from 'react';
import SelectorFilas from './SelectorFilas';

describe('Pruebas en el componente Accordion', () => {

  it('Debe de mostrarse correctamente el componente', () => {
    // arrange
    const onChange = jest.fn();
    const props = { rowsPerPage: 10, currentPage: 1, totalRows: 12, onChange };
    // act
    const wrapper = shallow(<SelectorFilas {...props} />);
    // assert
    expect(wrapper).toMatchSnapshot();
  });
});
