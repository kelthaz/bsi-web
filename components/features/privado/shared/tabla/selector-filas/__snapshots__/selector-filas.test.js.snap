// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente Accordion Debe de mostrarse correctamente el componente 1`] = `
<div
  className="inline"
>
  <span
    className="color-gray-light data-count"
  >
    1
     - 
    10
     de 
    12
     datos
  </span>
  <span
    className="color-gray py-1 ml-2 pl-2 rows-per-page"
  >
    Filas por p√°gina:
  </span>
  <div
    className="w-25 "
  >
    <Select
      blue={false}
      disabled={false}
      error=""
      inverted={false}
      items={
        Array [
          Object {
            "label": 5,
            "value": 5,
          },
          Object {
            "label": 10,
            "value": 10,
          },
          Object {
            "label": 20,
            "value": 20,
          },
          Object {
            "label": 50,
            "value": 50,
          },
        ]
      }
      label=""
      name="filasPorPagina"
      setTouched={[Function]}
      setValue={[Function]}
      showAlwaysErrors={false}
      size="smallb"
      touched={true}
      value={
        Object {
          "label": 10,
          "value": 10,
        }
      }
    />
  </div>
</div>
`;
